### ISSUE TITLE
Missing normalization context for any operation results in all schemas produced in output, including non-API resources

### ISSUE THREAD

**Issue by pauljura** [api-platform/api-platform#2203]
**API Platform version(s) affected**: 2.6.8

**Description**  

I have a bunch of Doctrine entities in my project, but I am only exposing 1 (so far) as an API Resource. I noticed that if I don't set "normalizationContext" at the top level, or a "normalization_context" on each operation, then the Swagger output will include **every** entity in the "Schemas" section, including entities that aren't marked as API Resources.

I understand that if I don't specify serialization groups then all properties of the entity will be exposed, that's fine. But I don't think it should expose every entity.

What's particularly odd to me is that a normalization context is even required for the DELETE operation, which ought to be unnecessary.

**How to reproduce**  

I want to allow only GET and DELETE operations, so I wrote the following attribute. This seems like it should be valid, in my opinion, but ended up exposing every entity in my project.

```
#[ApiResource(
    collectionOperations: [],
    itemOperations: [
        'get' => [
            'normalization_context' => ['groups' => ['member:read']],
        ],
        'delete',
    ],
)]
```

One way to avoid exposing every entity is by setting a default normalization context for all operations:

```
#[ApiResource(
    collectionOperations: [],
    itemOperations: [
        'get',
        'delete',
    ],
    normalizationContext: ['groups' => ['member:read']],
)]
```

Another way would be to set a normalization context on the DELETE operation, which I don't think should be required:

```
#[ApiResource(
    collectionOperations: [],
    itemOperations: [
        'get' => [
            'normalization_context' => ['groups' => ['member:read']],
        ],
        'delete' => [
            'normalization_context' => ['groups' => ['member:read']],
        ],
    ],
)]
```

And then of course if I don't specify any normalization context at all, then all entities are exposed.

In summary, this is easy to work around, but it confused me for a long time and doesn't seem like correct behaviour to me. And nothing in the documentation suggests that this is intentional.

**Comment by Ninos** [comment]
@mxmp210 using `ApiPlatform\Metadata\ApiResource` instead of `ApiPlatform\Core\Annotation\ApiResource` is not a proper solution here, because after that we cannot use the autogenerated `openapi_context` anymore (needs custom implementation).

Is it possible to add `normalizationContext/denormalizationContext` to `collectionOperations/itemOperations`? May also with autogenerated readOnly as already done for other properties with `#[ApiProperty]`?
This would solve a lot of basic restriction configuration problems in combination with openapi. Sometimes you just want to allow creating a property, but not updating it (easiest example: username in user entity class).

**Comment by soyuka** [comment]
Closing as stale per #2987 (no activity since 2022-10-23). This repository is transitioning to a CLI installer. If this still reproduces against a current release of API Platform, please open a new issue on `api-platform/core` with a minimal reproduction.
